src/main/java/seedu/address/model/module/Module.java

```diff
@@ -1,2 +1,127 @@
-package seedu.address.model.module;public class Module {
+package seedu.address.model.module;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import seedu.address.model.person.Professor;
+import seedu.address.model.person.Student;
+
+
+/**
+ * Represents a Module in the UniBook.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Module {
+
+    // Identity fields
+    private final ModuleName moduleName;
+    private final ModuleCode moduleCode;
+
+
+    // Data fields
+    private final Professor professor;
+    private final List<Student> students;
```

Comment:
Is List<Person> better here?

Reply:
List<Student> better
also make a List<Professor> 
as there can be more than 1 prof per mod

Reply:
thought about the observable list problem earlier, indeed we need the students and profs of a module to be in a observable list as well, else wont be able to track change in student/profs of a module from gui, so just use ObservableList<Student> and ObservableList<Professor>